bubble_sort_by.rb

```diff
@@ -0,0 +1,25 @@
+def bubble_sort_by(arr)
+  a = arr.length
+  while a.positive?
+    no_swaps = true
+    b = 0
+    while b < a - 1
+      left = arr[b]
+      right = arr[b + 1]
+      result = yield(left, right)
+      if result.positive?
+        arr[b], arr[b + 1] = arr[b + 1], arr[b]
+        no_swaps = false
+      end
+      b += 1
+    end
+    break if no_swaps
+
+    a -= 1
+  end
+  puts arr
+end
+
+bubble_sort_by(%w[hi hello hey]) do |left, right|
+  left.length - right.length
+end
```

Comment:
- [ ] Please delete all debugging code (all puts, prints, etc are debugging code if they serve to "test" your code). 
![image](https://user-images.githubusercontent.com/53256139/91252472-47a3a500-e723-11ea-99eb-ff7623b67d71.png)